src/BMPDecoder.ts

```diff
@@ -0,0 +1,79 @@
+import { IOBuffer } from 'iobuffer';
+
+import type { ImageCodec } from './BMPEncoder';
+
+export default class BMPDecoder {
+  bufferData: IOBuffer;
+  pixelDataOffset: number;
+  width: number;
+  height: number;
+  bitDepth: number;
+
+  constructor(bufferData: Buffer) {
+    this.bufferData = new IOBuffer(bufferData);
+    const formatCheck = this.bufferData.readBytes(2);
+    if (formatCheck[0] !== 0x42 && formatCheck[1] !== 0x4d) {
+      throw new Error(
+        'This is not a BMP image or the encoding is not correct.'
+      );
+    }
+    this.pixelDataOffset = this.bufferData.skip(8).readUint32();
+    this.width = this.bufferData.skip(4).readUint32();
+    this.height = this.bufferData.readUint32();
+    this.bitDepth = this.bufferData.seek(28).readUint16();
+    if (this.bitDepth !== 1) {
+      throw new Error('only bitDepth of 1 is supported');
+    }
+  }
+
+  decode(): ImageCodec {
+    this.bufferData.seek(this.pixelDataOffset);
+    this.bufferData.setLittleEndian();
+
+    let currentNumber = 0;
+    let currentWordIndex = 0;
+    let currentWord = 0;
+    let currentDataIndex = 0;
+    const rowSize = Math.floor((this.width + 31) / 32) * 4;
+
+    const data = new Uint8Array(Math.ceil((this.height * this.width) / 8));
+    this.bufferData.seek(this.pixelDataOffset);
+    this.bufferData.mark();
+    for (let row = this.height - 1; row >= 0; row--) {
```

Comment:
This code looks too complex.

Please explain me why you didn't use this simple approach that I sent you by Telegram ?

```
for (let row = 0; row < nbRows; row++) {
  for (let col = 0; col < nbCols; col++) {
    const bitIndex = col % 32;
    if (col % 32 === 0) {
      currentNumber = ioBuffer.readUint32();
    }
    if (currentNumber & (1 << bitIndex)) {
      // set pixel nbRows - row;
    } else {
      // clear pixel nbRows - row;
    }
  }
}
```